tests/test_all.py

```diff
@@ -742,6 +743,9 @@ def test_container_suseconnect_adds_repos(container_per_test: ContainerData):
     (SLE_BCI, SLE_BCI_debug and SLE_BCI_source).
 
     """
+    # Skip test on non-SLES hosts
+    if "SLES" not in LOCALHOST.check_output("cat /etc/os-release"):
+        pytest.skip("subscription tests must only run on SLES hosts")
```

Comment:
That's not what you want, what you actually want is to correct line 731 to:
```
not Path(ZYPP_CREDENTIALS_DIR).exists() or len(list(Path(ZYPP_CREDENTIALS_DIR).iterdir())) == 0
```
Yes, that's my fault. Then Leap should be skipped too, as long as it is unregistered.

My mistake was to only skip this test if /etc/zypp/credentials.d is absent, not when it is present, but empty

Reply:
I think that relying on the zypper credentials dir is not what we want because it makes the test prone to false negatives. If for some reason `/etc/zypp/credentials.d` is not present or created, the test would be wrongly skipped.

It's IMHO safer to explicitly state that this test MUST run on SLES and not rely on indirect assumptions that on SLES that directory must be present.